src/app/caesar-portal/group-list/create-edit-dialog/create-edit-dialog.component.ts

```diff
@@ -0,0 +1,287 @@
+        this.selectedBudgetOwner = this.budgetOwners[0];
+      },
+      error => console.log(error));
+  }
+
+  private showAllStages() {
+    this.updateGroupService.getStages().subscribe(
+      (data: Stage[]) => {
+        this.stages = data;
+        this.selectedStage = this.stages[0];
+      },
+      error => console.log(error));
+  }
+
+  public changeSelectedLocation(location) {
+    this.selectedLocation = location;
+    this.generateName();
+  }
+
+  public changeSelectedDirection(direction) {
+    this.selectedDirection = direction;
+    this.generateName();
+    this.calculateFinishDate();
+  }
+
+  public changeSelectedStage(stage) {
+    this.selectedStage = stage;
+  }
+
+  public changeSelectedBudgetOwner(budgetOwner) {
+    this.selectedBudgetOwner = budgetOwner;
+  }
+
+  public isActive(budgetOwner) {
+    return this.selectedBudgetOwner === budgetOwner;
+  }
+
+  // autogeneration group's name
+
+  public generateName () {
+    let result = '';
+    if (this.selectedLocation) {
+      result += `${this.locationNamesAlias[this.selectedLocation.id]}-`;
+      result += this.findLastInLocation(this.parseGroupsNames());
+      if (this.selectedDirection) {
+        result += `-${this.selectedDirection.name}`;
+      }
+    }
+    this.tmpGroupName = result;
+    this.groupName = result;
+    this.isShownSwitchName = false;
+  }
+
+  public parseGroupsNames (): Group[] {
+    const locationAlias = this.locationNamesAlias[this.selectedLocation.id],
+      pattern = new RegExp(locationAlias),
+      filteredGroups = this.groups.filter((value) => {
+        return value.name.match(pattern);
+      });
+    return filteredGroups;
+  }
+
+  public findLastInLocation (filteredGroups: Group[]): string {
+    let result = '',
+      tmpString = '',
+      lastGroup = filteredGroups[0];
+    for (let i = 0; i < filteredGroups.length; i++) {
+      if (filteredGroups[i].groupId > lastGroup.groupId) {
+        lastGroup = filteredGroups[i];
+      }
+    }
+    tmpString = lastGroup.name.substring(lastGroup.name.indexOf('-') + 1, lastGroup.name.lastIndexOf('-'));
+    if (Number(tmpString)) {
+      result += (Number(tmpString) + 1);
+    }
+    return result;
+  }
+
+  public switchGroupName () {
+    const gName = this.groupName,
+      tmpGName = this.tmpGroupName;
+    this.groupName = tmpGName;
+    this.tmpGroupName = gName;
+  }
+
+  public onNameModelChange(event) {
+    this.compareGeneratedGroupNames();
+  }
+
+  public compareGeneratedGroupNames () {
+    if (this.tmpGroupName !== this.groupName) {
+      this.isShownSwitchName = true;
+    }
+  }
+
+  // datepicker methods
+
+  public toggleDatepicker(shownDatepicker) {
+    this[shownDatepicker] = !this[shownDatepicker];
+  }
+
+  public selectionDone(shownDatepicker) {
+    this[shownDatepicker] = false;
+  }
+
+  public calculateFinishDate() {
+    if (this.dateStart) {
+      const start = new Date(this.dateStart),
+        shortDuration = 63,
```

Comment:
63, 84, 12, 14... I see a lot of magic numbers, what are they? Can you move them in some meaningful constants or add comments about them?